table.go

```diff
@@ -57,6 +59,96 @@ func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerst
 	return rt
 }
 
+// GetAllBuckets is safe to call as rt.Buckets is append-only
+// caller SHOULD NOT modify the returned slice
+func (rt *RoutingTable) GetAllBuckets() []*Bucket {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+	return rt.Buckets
+}
+
+// GenRandPeerID generates a random peerID in bucket=bucketID
+func (rt *RoutingTable) GenRandPeerID(bucketID int) (peer.ID, error) {
```

Comment:
https://github.com/aarshkshah1992/go-libp2p-kbucket/pull/1

Reply:
Also, let's just panic if the user passes a negative bucket. That's a program error, not a runtime error.

Reply:
@Stebalien  Thanks for the beautiful bit magic. Just had to make some changes to it. Have explained them here #r317347278